src/framework/ui/view/interactiveprovider.cpp

```diff
@@ -175,19 +177,35 @@ Ret InteractiveProvider::showProgress(const std::string& title, Progress* progre
 RetVal<io::path_t> InteractiveProvider::selectOpeningFile(const std::string& title, const io::path_t& dir,
                                                           const std::vector<std::string>& filter)
 {
-    return openFileDialog(FileDialogType::SelectOpenningFile, title, dir, filter);
+    RetVal<InteractiveProvider::FileInfo> fileInfo = openFileDialog(FileDialogType::SelectOpenningFile, title, dir, filter);
+    return fileInfo.ret.code()
+           == int(Ret::Code::Ok) ? RetVal<io::path_t>::make_ok(fileInfo.val.path) : RetVal<io::path_t>(fileInfo.ret);
```

Comment:
Can be written shorter as 
```suggestion
    return fileInfo.ret ? RetVal<io::path_t>::make_ok(fileInfo.val.path) : fileInfo.ret;
```
(similar in `InteractiveProvider::selectDirectory`)

Reply:
Fixed! Thank you!